lib/resque/plugins/disable_job/job.rb

```diff
@@ -93,29 +92,23 @@ def self.remove_specific_rule(rule)
         # To set the arguments to block we need to keep in mind the parameter order and that
         # if we don't specify anything, that means we are blocking everything.
         # The rule is from generic to specific.
-        def self.args_match(args, set_args)
-          return true if args == set_args
-          should_block = args.to_a.map.with_index do |a, i|
-            # We check each parameter (65) or parameters set (["account_id", 65]) in the job_args with the job_args to be blocked
-            # if it's nil, then we match, if it's specified, we check for equality (65 == 65 or ["account_id", 65] == ["account_id", 65])
-            set_args[i].nil? || a == set_args[i]
+        def self.args_match(job_args, set_args)
+          return true if job_args == set_args
+          # We check each parameter in the job_args with the job_args to be blocked
+          # if it's nil, then we match, if it's specified, we check for equality (65 == 65)
+          should_block = if set_args.is_a?(Hash)
+            # If the set arguments is a hash we try to extract the job arguments as a hash
+            job_args = job_args.size == 1 && job_args.first.is_a?(Hash) ? job_args.first : job_args
+            job_args.map { |k, v| set_args[k].nil? || v == set_args[k] }
+          else
+            job_args.map.with_index { |a, i| set_args[i].nil? || a == set_args[i] }
```

Comment:
These comments are a smell IMO, can you extract an object to encapsulate all the checking that's going on here?